test/Interop/Cxx/reference/const-ref-parameter.swift

```diff
@@ -0,0 +1,77 @@
+// REQUIRES: objc_interop
+
+import ConstRefParameter
+
+func testFunction() {
+    let a = OptionsStruct(intOption: 1, floatOption: 2.0)
+
+    let objc = OptionsConsumerObjC(options: a)
+    _ = objc.doOtherThing(withOptions: a)
+    _ = OptionsConsumerObjC.consumer(withOptions: a)
+    _ = OptionsConsumerObjC.doThing(withOptions: a)
+
+    var cxx = OptionsConsumerCxx(a)
+    _ = cxx.doOtherThing(a)
+    _ = OptionsConsumerCxx.build(a)
+    _ = OptionsConsumerCxx.doThing(a)
+}
+
+// RUN: %target-swift-ide-test -print-module -module-to-print=ConstRefParameter -I %S/Inputs -source-filename=x -enable-experimental-cxx-interop -enable-objc-interop | %FileCheck -check-prefix=CHECK-IDE-TEST %s
+
+// CHECK-IDE-TEST: class OptionsConsumerObjC
+// CHECK-IDE-TEST: init(options: OptionsStruct)
+// COM: FIXME: should it be consumer(options:)?
```

Comment:
I don't see why?

Reply:
According to https://github.com/apple/swift/blob/main/docs/CToSwiftNameTranslation.md#factory-initializers for a factory initializer the "with" should be dropped. I don't know why it is not happening here.

But also, there's a phrase that says "If a factory initializer turns out to have the same imported name as an available designated initializer, or a non-inherited factory initializer has the same imported name as an available convenience initializer, the factory initializer is marked unavailable to resolve ambiguities." which also should had happen here, but it doesn't (`consumer` should have turn into `init`).

Something is wonky here, but I am not sure what exactly is failing.